README.md

```diff
@@ -1,3 +1,339 @@
+
+Shows the description from the metadata file
+
+#### `list`
+
+Shows all available commands
+
+#### `self-update`
+
+Attempts to upgrade the npecl tool itself
+
+#### `show`
+
+Shows all installed extensions available with the PHP version in the path, including their versions. This includes
+*all* loaded PHP extensions, and not just npecl ones.
+
+#### `verify {?version-constraint}`
+
+Shows the signature of the person who signed the version npecl would install, or from the specific version, if given.
+
+#### `upgrade {?{ext-name}{?:version-constraint}{?@dev-branch-name}}`
+
+Attempts to upgrade all installed versions to the latest available ones on GitHub (unless their major version has
+changed) If a specific package is specified, only attempt to upgrade that specific one.
+
+### CLI Options
+
+#### `--dry-run`
+
+Shows what it is about to do, but doesn't actually install
+
+#### `--force`
+
+To attempt to install a version that doesn't match the version constraints from the meta-data, for instance to install
+an older version than recommended, or when the signature is not available.
+
+#### `--pre-release`
+To allow for installation of pre-release versions
+
+#### `--with-php-config=/path/to/php-config`
+
+Allows installation of extensions with PHP versions that are not in the path
+
+#### `--{option}{?=value}`
+
+All options specified in the `config` section of the composer.json file can also be given, including a value if they
+take them. For example, for Xdebug you could run `npecl install xdebug --without-xdebug-compression`.
+
+## Extension maintainer: register an npecl package
+
+Create a `composer.json` in your repository, commit:
+
+```json
+{
+    "name": "ext-xdebug",
+    "type": "php-ext-zend",
+    "license": "Xdebug-1.03",
+    "description": "Xdebug is an amazing tool...",
+    "require": {
+        "ext-zlib": "*",
+        "php": ">=7.4.21,<8.4"
+    },
+    "conflict": {
+        "ext-zetalib": "*"
+    },
+    "php-ext": {
+        "priority": 90,
+        "config": {
+            "enable-xdebug-dev": false,
+            "without-xdebug-compression": false
+        }
+    }
+}
+```
+
+Submit your repo URL into [https://packagist.org/packages/submit](https://packagist.org/packages/submit). Packagist
+will read the `composer.json` as usual.
+
+When releases are made, Packagist will do the usual thing of producing metadata for the release
+
+### Windows binaries
+
+Windows needs pre-built binary DLLs. The expected workflow is that the release is made, then some kind of build takes
+place (for example, in a GitHub Action, or manually on a compatible build environment), and the `.dll` file is added to
+the GitHub release.
+
+The name for the DLLs must follow the following pattern:
+
+ * `php_{extension-name}-{tag}-{php-maj/min}-{compiler}{"-nts"}?{-arch}?.dll`
+
+The descriptions of these items:
+
+ * `extension-name` the name of the extension, e.g. `xdebug` - this is the same as defined in `composer.json` minus `ext-`
```

Comment:
Does it make sense to have an obligatory organization prefix in the extension name, I.e., `ext-xdebug-xdebug`, or `ext-symfony-console` for non-official extension? It may prepare for possible conflicts in the future, where some extensions could be incorporated into the PHP core.

Reply:
While I am a huge fan of the decision we made for packagist.org to enforce vendor names, I think in this case I'd say it's not needed because of a few reasons:

- BC with existing names
- The total number of exts is 431 on pecl, and I suspect a bunch are historical and will never even be added to the new listing. In contrast packagist.org currently lists almost 400K packages, making the name clashes much more problematic.
- If an ext moves to php core, then you ideally want the name to stay the same IMO, it just moves org/ownership but would be represented to Composer as ext-foo before *and* after the transition.

Reply:
I should note, that there is nothing stopping an extension developer calling their extension `symfony-console` to use one of your examples, but I agree with @Seldaek that namespacing is not likely to be needed, given the limited specialist needs that warrant extensions to exist.